### DISCUSSION TITLE
Change time in logs wg-easy

### DISCUSSION THREAD

**Discussion by zingarini** [wg-easy/wg-easy#766] [Q&A]
Time in the wg-easy logs corect for UTC, which is different from my TZ. Other containers display the time correctly.
Ubuntu 22.04, timezone configured. /etc/localtime & /etc/timezone exported to docker.
Perhaps there are ideas on how to fix TZ?
Thanks!

**Comment by peterlewis** [comment]
Adding `TZ=WhateverYouWant` to environment variables should do the trick.

**Reply by zingarini** [reply]
Adding TZ to environment variables I tried first. As a standalone container, i mean docker run -e TZ=WhateverIWant
and as a compose, like:
   wg-easy:
    environment:
      - TZ=WhateverIWant
Adding environment variables works with any other containers correctly except wg-easy :(

**Reply by pheiduck** [reply]
Add this
`- /etc/localtime:/etc/localtime:ro`
under `volumes:`

**Reply by pheiduck** [reply]
Indeed it's 1h behind my localtime

**Reply by pheiduck** [reply]
> Add this `- /etc/localtime:/etc/localtime:ro` under `volumes:`

Does not work either.

**Reply by eandersons** [reply]
I did not test this, but I have seen mounting `/etc/timezone` into a container along with `/etc/localtime`. If operating system does not have such file, it can be created in any directory and it should contain [zone Id](https://nodatime.org/TimeZones) (for example, `Europe/Copenhagen`).

**Reply by zingarini** [reply]
Thanks for support!
However, as we wrote earlier, mounting /etc/timezone and /etc/localtime into a container does not solve the problem.
I guess the problem is that the image is Alpine-based. 
Accordingly, the issue can certainly be resolved by assembling an image with the addition someone like this:
FROM alpine:3.6
RUN apk add --no-cache tzdata
ENV TZ America/Los_Angeles

**Reply by pheiduck** [reply]
@zingarini please test this dev build: `ghcr.io/wg-easy/wg-easy:development`
https://github.com/wg-easy/wg-easy/commit/8bbdce39904c48ce0327e418d19b1ac2da494cdf

**Reply by pheiduck** [reply]
Time is changed inside container but it does not work for logs on my side.

**Reply by pheiduck** [reply]
With Watchtower it works as I wrote earlier. I need to find out how they did it.

**Reply by zingarini** [reply]
> Time is changed inside container but it does not work for logs on my side.

exactly. But I decided that this is incorrect configuration of Docker on my host.

**Reply by pheiduck** [reply]
PR is in WIP: https://github.com/wg-easy/wg-easy/pull/784
but might be not included in next update.
